test/functional/legacy/assert_spec.lua

```diff
@@ -233,20 +234,31 @@ describe('assert function:', function()
   -- assert_fails({cmd}, [, {error}])
   describe('assert_fails', function()
     it('should change v:errors when error does not match v:errmsg', function()
-      command([[call assert_fails('xxx', {})]])
+      eq(1, eval([[assert_fails('xxx', {})]]))
       command([[call assert_match("Expected {} but got 'E731:", v:errors[0])]])
       expected_errors({"Expected {} but got 'E731: using Dictionary as a String'"})
     end)
 
     it('should not change v:errors when cmd errors', function()
-      call('assert_fails', 'NonexistentCmd')
+      eq(0, eval([[assert_fails('NonexistentCmd')]]))
       expected_empty()
     end)
 
     it('should change v:errors when cmd succeeds', function()
-      call('assert_fails', 'call empty("")')
+      eq(1, eval([[assert_fails('call empty("")', '')]]))
       expected_errors({'command did not fail: call empty("")'})
     end)
+
+    it('can specify and get a message about what failed', function()
+      eq(1, eval([[assert_fails('xxx', {}, 'stupid')]]))
+      command([[call assert_match("stupid: Expected {} but got 'E731:", v:errors[0])]])
+      expected_errors({"stupid: Expected {} but got 'E731: using Dictionary as a String'"})
+    end)
+
+    it('can specify and get a message even when cmd succeeds', function()
+      eq(1, eval([[assert_fails('echo', '', 'echo command')]]))
+      expected_errors({'command did not fail: echo command'})
+    end)
```

Comment:
Shouldn't we go with the upstream src/testdir/test_assert.vim ?

Reply:
Yes, but there are 8.1.1xxxx method patches for assert functions so test_assert.vim will be synced with upstream version eventually.

Reply:
@janlazo 
Can you explain what you've meant?

I thought the `src/testdir/test_assert.vim` tests would be enough, and we do not need the legacy test in Lua then in the first place.
Did you mean that currently things are missing for it currently?

(but then it might have been better to merge those first)

Reply:
`test_assert.vim` is enough and shouldn't have been ported over as a lua test. There are future 8.1.x patches that add tests to `test_assert.vim` so deleting `assert_spec.lua` can be done in the same vim-patch PR.

I didn't want to do it in this PR because `test_assert.vim` changes weren't ported in order.